internal/controller/authenticator.go

```diff
@@ -133,6 +136,8 @@ func NewClientSet() (*kubernetes.Clientset, error) {
 		return nil, err
 	}
 
+	clusterConfig.Proxy = http.ProxyFromEnvironment
```

Comment:
Why did you revert this?
Cf. my previous comment: https://github.com/illumio/cloud-operator/pull/195#discussion_r2067943714
```suggestion
	// Do not use the configured HTTP proxy for connections to apiserver within this cluster,
	// since it is only meant to be used for connections to CloudSecure
	clusterConfig.Proxy = http.ProxyURL(nil)
```
The `Proxy` field already defaults to `http.ProxyFromEnvironment`.
The problem with this default setting is that the HTTPS proxy configured for cloud-operator is meant to be used only for the egress connections from cloud-operator to Cloud Secure. It doesn't make sense to use it for communication between this pod and the apiserver within the same cluster. This is surely going to fail.